### PR TITLE
#160609556 rate an article

### PR DESCRIPTION
**Pull request by mendozabree**
#### What does this PR do?

Authenticated users can rate an article
#### Description of Task to be completed?

Currently, authenticated users cannot rate an article.

This task allows authenticated users to rate an article.

#### How should this be manually tested?

Make migrations:  `./manage.py makemigrations articles`
Migrate: `./manage.py migrate`
Run the server: `./manage.py runserver`
Open postman and test these endpoints: **Note**; All endpoints are authenticated

**Rate an article**
This is the request body:
```
{“rate”: 4}
```
POST `http://127.0.0.1:8000/api/articles/<slug>/rate/`

#### What are the relevant pivotal tracker stories?

[#160609556](https://www.pivotaltracker.com/story/show/160609556)

#### Any background context you want to add?
N/A

#### Screenshots

### CODE REVIEW

**Review by samuelbwambale — COMMENTED** [review]
@RonKbS There are some merge conflicts in one file `urls.py`. Please fix them before the PR can be merged

File: authors/apps/articles/serializers.py [hunk 1]
```diff
@@ -150,7 +153,27 @@ def create(self, validated_data):
         )
 
 
-
 class ShareArticleSerializer(serializers.Serializer):
     content = serializers.CharField()
     share_with = serializers.CharField()
+class RateSerializer(serializers.ModelSerializer):
```
Comment (MuhanguziDavid) [inline authors/apps/articles/serializers.py:159]: @mendozabree , please add two lines before this new class.

File: authors/apps/articles/views.py [hunk 2]
```diff
@@ -500,3 +504,42 @@ def send_mail(sender_email, receiver_mail, mail_subject, content):
     mail = Mail(from_email, subject, to_email, content)
     response = sg.client.mail.send.post(request_body=mail.get())
     return response
+class RateView(APIView):
+    permission_classes = (IsAuthenticated,)
+    serializer_class = RateSerializer
+
+    def average_rating(self, article):
+
```
Comment (MuhanguziDavid) [inline authors/apps/articles/views.py]: Please remove this redundant line.

Comment (mendozabree) [reply authors/apps/articles/views.py]: Thanks for pointing that out, I have worked on it!

File: authors/apps/articles/serializers.py (hunk 1 above)
Comment (mendozabree) [reply authors/apps/articles/serializers.py:159]: Thanks for pointing that out, I have worked on it!

**Review by MuhanguziDavid — APPROVED** [review]
LGTM

**Review by samuelbwambale — APPROVED** [review]
Changes approved. LGTM

File: authors/apps/articles/tests.py [hunk 3]
```diff
@@ -390,12 +395,15 @@ def setUp(self):
         self.response = self.client.post(
             '/api/users/', self.user_data, format="json"
         )
+
         self.client.credentials(
             HTTP_AUTHORIZATION='Token {}'.format(self.response.data['auth_token'])
-        )
+        )
```
Comment (kakaemma) [inline authors/apps/articles/tests.py:401]: @mendozabree , kindly remove the extra spaces here.

File: authors/apps/articles/tests.py [hunk 4]
```diff
@@ -552,3 +560,70 @@ def test_thread_creation_with_another_thread_as_parent(self):
             "Parent comment is already a sub comment",
             str(result.data['message'])
         )
+
```
Comment (kakaemma) [inline authors/apps/articles/tests.py:563]: Consider adding another blank line here to follow the two blank lines before a class.

**Pull request merged**